src/SceneBinding.ts

```diff
@@ -537,16 +544,41 @@ class SceneBinding {
   };
 
   private createNode_ = async (id: string, node: Node, nextScene: Scene): Promise<Babylon.Node> => {
+    let nodeToCreate: Node = node;
+
+    // Resolve template nodes into non-template nodes by looking up the template by ID
+    if (node.type === 'from-template') {
+      const nodeTemplate = preBuiltTemplates[node.templateId];
+      if (!nodeTemplate) {
+        console.warn('template node has invalid template ID:', node.templateId);
+        return null;
+      }
+
+      nodeToCreate = {
+        ...node,
+        ...nodeTemplate,
```

Comment:
is the order here right? Shouldn't the node overwrite whatever is in the template?